File: demo/NUnitTestDemo/ExpectedOutcomeAttributes.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using NUnit.Framework;

namespace NUnitTestDemo
{
    public class ExpectPassAttribute : PropertyAttribute
    {
        public ExpectPassAttribute() : base("Expect", "Pass") { }
    }


    public class ExpectFailureAttribute : PropertyAttribute
    {
        public ExpectFailureAttribute() : base("Expect", "Failure") { }
    }

    public class ExpectWarningAttribute : PropertyAttribute
    {
        public ExpectWarningAttribute() : base("Expect", "Warning") { }
    }

    public class ExpectIgnoreAttribute : PropertyAttribute
    {
        public ExpectIgnoreAttribute() : base("Expect", "Ignore") { }
    }

    public class ExpectSkipAttribute : PropertyAttribute
    {
        public ExpectSkipAttribute() : base("Expect", "Skipped") { }
    }

    public class ExpectErrorAttribute : PropertyAttribute
    {
        public ExpectErrorAttribute() : base("Expect", "Error") { }
    }

    public class ExpectInconclusiveAttribute : PropertyAttribute
    {
        public ExpectInconclusiveAttribute() : base("Expect", "Inconclusive") { }
    }

    public class ExpectMixedAttribute : PropertyAttribute
    {
        public ExpectMixedAttribute() : base("Expect", "Mixed") { }
    }
}

File: demo/vb/ExpectedOutcomeAttributes.vb
Imports NUnit.Framework

Namespace NUnitTestDemo

    Public Class ExpectPassAttribute
        Inherits PropertyAttribute

        Public Sub New()
            MyBase.New("Expect", "Pass")
        End Sub

    End Class


    Public Class ExpectFailureAttribute
        Inherits PropertyAttribute

        Public Sub New()
            MyBase.New("Expect", "Failure")
        End Sub

    End Class

    Public Class ExpectWarningAttribute
        Inherits PropertyAttribute

        Public Sub New()
            MyBase.New("Expect", "Warning")
        End Sub

    End Class

    Public Class ExpectIgnoreAttribute
        Inherits PropertyAttribute

        Public Sub New()
            MyBase.New("Expect", "Ignore")
        End Sub

    End Class

    Public Class ExpectSkipAttribute
        Inherits PropertyAttribute

        Public Sub New()
            MyBase.New("Expect", "Skipped")
        End Sub

    End Class

    Public Class ExpectErrorAttribute
        Inherits PropertyAttribute

        Public Sub New()
            MyBase.New("Expect", "Error")
        End Sub

    End Class

    Public Class ExpectInconclusiveAttribute
        Inherits PropertyAttribute

        Public Sub New()
            MyBase.New("Expect", "Inconclusive")
        End Sub

    End Class

    Public Class ExpectMixedAttribute
        Inherits PropertyAttribute

        Public Sub New()
            MyBase.New("Expect", "Mixed")
        End Sub

    End Class

End Namespace